spec/takeaway_spec.rb

```diff
@@ -0,0 +1,124 @@
+require 'takeaway'
+
+describe Takeaway do
+  describe '#new_order'
+
+    it "until a new_order, menu is nil" do
+      expect(subject.menu).to eq nil
+    end
+
+  let(:takeaway) { described_class.new(restaurant) }
+  let(:restaurant) { double() }
+  let(:cuisine) { double(menu: {item: 5}) }
+
+
+    it "when new_order is started, menu is populated with a specific cuisine" do
+      takeaway.new_order(cuisine)
+      expect(takeaway.menu).to eq cuisine.menu
+    end
+
+  before do
+    allow(restaurant).to receive(:new).and_return(cuisine)
+    takeaway.new_order(cuisine)
+    takeaway.add(:item, 5)
+  end
+
+  context "add item(s)" do
+
+    it "with default quantity of 1" do
+      expect { takeaway.add(:item) }.to change { takeaway.order[:item] }.by(1)
+    end
+
+    it "with explicit quantity" do
+      expect { takeaway.add(:item, 2) }.to change { takeaway.order[:item] }.by(2)
+    end
+
+    it "but item does not exist" do
+      expect(takeaway.add(:no_item)).to eq "Sorry, your selection is unavailable"
+    end
+
+  end
+
+  context "delete item(s)" do
+
+    it "with default quantity of 1" do
+      expect { takeaway.delete(:item) }.to change { takeaway.order[:item] }.by(-1)
```

Comment:
Metrics/LineLength: Line is too long. [82/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)